external/js/cothority/src/network/proto.ts

```diff
@@ -251,15 +218,28 @@ export class ServerIdentity extends Message<ServerIdentity> {
     }
 
     /**
-     * Returns websocket version of this.url if set, otherwise converts the server
-     * address to match the websocket format.
+     * Returns websocket URL to connect to this ServerIdentity.
      * @returns the websocket address
      */
     getWebSocketAddress(): string {
-        if (this.url) {
-            return ServerIdentity.urlToWebsocket(this.url);
+        if (this.url !== undefined && this.url !== "") {
+            const url = new URL(this.url, {});
+            // TODO why not supporting proto ws: & wss: ?
+            switch (url.protocol) {
+                case "http:": url.set("protocol", "ws:"); break;
+                case "https:": url.set("protocol", "wss:"); break;
+                default: throw new Error(`unknown url protocol: ${url.protocol}`);
+            }
+            return url.href;
         } else {
-            return ServerIdentity.addressToWebsocket(this.address);
+            const addr = new URL(this.address, {});
+            const port = parseInt(addr.port, 10);
+            if (isNaN(port)) {
```

Comment:
```javascript
Number.isNaN(port)
```
The global tries to cast the value to a number but it will always be a number/NaN from parseInt function